sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -970,25 +966,24 @@ class Analyzer(
         tableName: Seq[String],
         changes: Seq[TableChange]): Option[AlterTable] = {
       lookupV2Relation(tableName) match {
-        case scala.Left((v2Catalog, ident, tableOpt)) =>
+        case (Some(table), Some((catalog, ident))) =>
           Some(AlterTable(
-            v2Catalog.asTableCatalog,
+            catalog,
             ident,
-            tableOpt.map(DataSourceV2Relation.create).getOrElse(UnresolvedRelation(tableName)),
-            changes
-          ))
-        case scala.Right(tableOpt) =>
-          tableOpt.map { table =>
-            AlterTable(
-              sessionCatalog.asTableCatalog,
-              Identifier.of(tableName.init.toArray, tableName.last),
-              DataSourceV2Relation.create(table),
-              changes
-            )
-          }
+            DataSourceV2Relation.create(table),
+            changes))
+        case (Some(table), None) =>
+          Some(AlterTable(
+            sessionCatalog.asTableCatalog, // table being resolved means this exists
```

Comment:
maybe use `catalogManager.v2SessionCatalog` to be consistent with `lookupV2Relation`?